packages/api_bo/src/resolvers/instructors.ts

```diff
@@ -405,5 +417,126 @@ export const instructors = {
         throw new Error("500, " + error);
       }
     },
+
+    setStatusInstructor: async (
+      _parent: unknown,
+      args: MutationSetStatusInstructorArgs,
+      ctx: Context,
+    ): Promise<InstructorModel> => {
+      try {
+        checkNotNull(args);
+        let instructor = await instructorCollection(ctx.db).findAndModify(
+          { _id: new ObjectId(args.id) },
+          {
+            update: { $set: { enrolled: args.enrolled } },
+            new: true,
+          },
+        ) as InstructorModel;
+        if (!instructor) {
+          throw new Error("404, Instructor not found");
+        }
+
+        //check if center of groups are active
+        const groups = await groupCollection(ctx.db).aggregate<
+          { _id: ObjectId }
+        >([
+          {
+            $lookup: {
+              from: "centers",
+              localField: "center",
+              foreignField: "_id",
+              as: "center",
+            },
+          },
+          {
+            $match: {
+              instructors: new ObjectId(args.id),
+              "center.active": true,
+            },
+          },
+          {
+            $project: {
+              _id: 1,
+            },
+          },
+        ]).toArray();
+
+        const idsActiveCenterGroups = groups.map((item) => item._id);
+
+        const activeCenterGroups = await groupCollection(ctx.db).find({
+          _id: { $in: idsActiveCenterGroups },
+        }).toArray();
+
+        if (activeCenterGroups.length > 0) {
+          if (!args.enrolled) {
+            instructor = await instructorCollection(ctx.db).findAndModify(
+              { _id: new ObjectId(args.id) },
+              {
+                update: { $set: { corporateEmail: null } },
+                new: true,
+              },
+            ) as InstructorModel;
+
+            //if groups have all inactive instructors, set group active to false
+            const updateGroups = await Promise.all(
+              activeCenterGroups.map(async (group) => {
+                const instructors = await instructorCollection(ctx.db)
+                  .countDocuments({
+                    _id: { $in: group.instructors },
+                    enrolled: true,
+                  });
+                if (instructors === 0) {
+                  return group._id;
+                }
+              }),
+            ) as ObjectId[];
```

Comment:
creo que queda más limpio tipar la variable al lado del nombre que hacer un `as`

Reply:
me da este error cuadno tipo: Type '(ObjectId | undefined)[]' is not assignable to type 'ObjectId[]'

Reply:
Pero en este caso es verdad, no? puede que haya casos en los que no haya ningún monitor y en las que devuelva `undefined`

Reply:
He quitado todos los as que he podido

Reply:
lo que devuelve updateGroups son los grupos en los que no se encuentra ningún monitor activo y luego utilizo ese array para modificar esos grupos a active = false. Entonces cuando hago: 
await groupCollection(ctx.db).updateMany({
              _id: { $in: updateGroups },
            }, { $set: { active: false } }); me dice que ese updateGroups tiene que ser un ObjectId[] por eso lo he updateGroups con as

Reply:
Vale, guay :)
En general cuando te den errores de tipos de mongo creo que es mejor tipar el retorno del update o lo que estés haciendo (`collection.update<ReturnType>(....)`) antes que poner el as.